devicetypes/smartthings/zigbee-motion-sensor-light.src/led-cpx-light.groovy

```diff
@@ -0,0 +1,166 @@
+		}
+		
+		controlTile("colorTempSliderControl", "device.colorTemperature", "slider", width: 4, height: 2, inactiveLabel: false, range: "(2700..6500)") {
+			state "colorTemperature", action: "color temperature.setColorTemperature"
+		}
+		
+		main(["switch"])
+		details(["switch", "switchLevel", "colorTempSliderControl", "refresh"])
+	}
+}
+
+private getMOTION_CLUSTER_VALUE() { 0x0406 }
+private getMOTION_STATUS_VALUE() { 0x0000 }
+private getON_OFF_CLUSTER_VALUE() { 0x0006 }
+private getON_OFF_COMMAND_VALUE() { 0x07 }
+
+def parse(String description) {
+	def event = zigbee.getEvent(description)
+	
+	if (event) {
+		if (!(event.name == "level" && event.value == 0)) {
+			sendEvent(event)
+		}
+	} else {
+		def cluster = zigbee.parse(description)
+		
+		if (cluster && cluster.clusterId == ON_OFF_CLUSTER_VALUE && cluster.command == ON_OFF_COMMAND_VALUE) {
+			if (cluster.data[0] == 0x00) {
+				sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+			}
+		} else {
+			def zigbeeMap = zigbee.parseDescriptionAsMap(description)
+			
+			if (zigbeeMap.clusterInt ==  MOTION_CLUSTER_VALUE && zigbeeMap.attrInt == MOTION_STATUS_VALUE) {
+				def childDevice = getChildDevices()?.find {
+					it.device.deviceNetworkId == "${device.deviceNetworkId}:1" 
+				}
+				zigbeeMap.value = zigbeeMap.value.endsWith("01") ? "active" : "inactive"			
+				zigbeeMap.name = "motion"
+				childDevice.sendEvent(zigbeeMap)
+			}
+		}
+	}
+}
+
+def off() {
+	zigbee.off()
+}
+
+def on() {
+	zigbee.on()
+}
+
+def setLevel(value, rate=null) {
+	zigbee.setLevel(value)
+}
+
+def configure() {
+	zigbee.configureReporting(MOTION_CLUSTER_VALUE, MOTION_STATUS_VALUE, 0x18, 30, 600, null) +
+		zigbee.onOffConfig() +
+		zigbee.levelConfig() +
+		refresh()
+}
+
+def updated() {
+	refresh()
+}
+
+def ping() {
+	return zigbee.levelRefresh()
+}
+
+def refresh() {
+	zigbee.readAttribute(MOTION_CLUSTER_VALUE, MOTION_STATUS_VALUE) +
+		zigbee.onOffRefresh() +
+		zigbee.levelRefresh() +
+		zigbee.colorTemperatureRefresh()
+}
+
+def setColorTemperature(value) {
+	value = value as Integer
+	
+	zigbee.setColorTemperature(value) +
+		zigbee.on() +
+		zigbee.colorTemperatureRefresh()
+}
+
+def installed() {
+	addChildSensor()
+}
+
+def addChildSensor() {
+	def componentLabel
+	
+	if (device.displayName.endsWith(' Light') || device.displayName.endsWith(' light')) {
+		componentLabel = "${device.displayName[0..-6]} Motion sensor"
+	} else {
+		componentLabel = "$device.displayName Motion sensor"
+	}
+	
+	try {
+		String dni = "${device.deviceNetworkId}:1"
+		addChildDevice("ITM CPX Motion sensor child", dni, device.hub.id, [completedSetup: true, label: "${componentLabel}", isComponent: false])
+	} catch (e) {
+		log.warn "Failed to add ITM Fan Controller - $e"
+	}
+	
+	if(childDevice != null) {
+		childDevice.sendEvent(name: "motion", value: "inactive")
+	}
```

Comment:
Here `childDevice` will be definitely null, as it isn't even declared. You need to declare it before `try {} catch {}` block, and then assign to it returned value from `addChildDevice`, as discussed earlier.
Also, missing space after `if`.

Reply:
I wonder if I edited that correctly.